src/components/common/basic/CommonTimeline.vue

```diff
@@ -0,0 +1,1048 @@
+    },
+    scale: {
+        type: Number as PropType<number>,
+        required: false,
+        default: 1,
+    },
+    collapsed: {
+        type: Boolean as PropType<boolean>,
+        required: false,
+    },
+    utc: {
+        type: Boolean as PropType<boolean>,
+        required: false,
+    },
+});
+
+const isMobile = useIsMobile();
+
+const ready = ref(false);
+
+const cellWidth = isMobile.value ? 80 : 120;
+const rowHeight = 50;
+const gap = 4;
+const headerWidth = props.headers.length * cellWidth;
+
+// I dont care if its reactivity is losed....didnt found a workaround, because toRef cannot be edited ... so its like this :) suggestions?
+// eslint-disable-next-line vue/no-setup-props-reactivity-loss
+const scale = ref(props.scale);
+
+const isDragging = ref(false);
+const startX = ref(0);
+const startY = ref(0);
+const dragContainer = ref<HTMLElement | null>(null);
+
+const scrollSpeed = 5;
+const scrollDown = ref(0);
+const scrollUp = ref(0);
+const scaleIngrement = ref(0.5);
+
+const getEntries = toRef(props, 'entries');
+const getIdentifiers = toRef(props, 'identifiers');
+
+const idContainers = ref<{ [key: string]: HTMLElement | null }>({});
+const idTexts = ref<{ [key: string]: HTMLElement | null }>({});
+const collapsedEntries = ref<Map<number, TimelineEntry[]>>(new Map());
+const indexOffsetMap = ref<Map<number, number>>(new Map());
+const collapseMap = ref(new Map<string, boolean>());
+
+const endCalculated = computed(() => {
+    if (props.end) {
+        return props.end;
+    }
+
+    let tmp = new Date();
+    getEntries.value.forEach(entry => {
+        if (entry.end > tmp) {
+            tmp = new Date(entry.end.getTime());
+        }
+    });
+
+    return tmp;
+});
+
+const currentMinute = ref(new Date());
+
+const now = computed(() => currentMinute.value);
+const utc = toRef(props, 'utc');
+const timeZone = computed(() => utc.value ? 'UTC' : undefined);
+
+const formatterTime = computed(() => new Intl.DateTimeFormat(['de-DE'], {
+    hour: '2-digit',
+    minute: '2-digit',
+    timeZone: timeZone.value,
+}));
+
+const formatterDate = computed(() => new Intl.DateTimeFormat(['de-DE'], {
+    day: '2-digit',
+    month: '2-digit',
+    timeZone: timeZone.value,
+}));
+
+const idsRef = ref(prepareIds());
+
+const entriesRef = ref(filterEntries());
+watch([getEntries, getIdentifiers], () => {
+    collapsedEntries.value.clear();
+    indexOffsetMap.value.clear();
+    idsRef.value = prepareIds();
+    entriesRef.value = filterEntries();
+
+    collapseByMap();
+});
+
+const getTimeline: Ref<TimelineTime[]> = shallowRef(generateTimeline());
+const getTimelineDays: Ref<TimelineTime[]> = shallowRef(generateTimelineDays());
+
+watch(scale, () => {
+    getTimeline.value = generateTimeline();
+    getTimelineDays.value = generateTimelineDays();
+});
+
+const flattenedCollapsedEntries = computed(() => Array.from(collapsedEntries.value.values()).flatMap(entries => entries));
+
+onMounted(() => {
+    setInterval(() => {
```

Comment:
Memory leak unless you clear it

```suggestion
    const interval = setInterval(() => {
```

And use onBeforeUnmount below to clearInterval